lib/ruby_lsp/executor.rb

```diff
@@ -89,9 +89,10 @@ def run(request)
         return cached_response if cached_response
 
         # Run listeners for the document
-        document_symbol = Requests::DocumentSymbol.new(uri, @message_queue)
-        document_link = Requests::DocumentLink.new(uri, @message_queue)
-        EventEmitter.new(document_symbol, document_link).visit(document.tree) if document.parsed?
+        emitter = EventEmitter.new
+        document_symbol = Requests::DocumentSymbol.new(emitter, @message_queue)
```

Comment:
For consistency, might it help to always pass the `uri` if not needed for some listeners?

Reply:
That's part of what I'm trying to avoid. Some requests will require position, others range or URI. At the end of migrating, we'll need 5 or 6 positional arguments for every listener even if they don't require these arguments.

I think it's best to not have consistency in this case and just let each listener receive the relevant arguments.

Reply:
I'm not sure I agree. There will always be consistency for all listeners for the same request type. For example, all document symbol listeners are instantiated the same.

However, trying to enforce that different requests have the same listener arguments is very limiting and I can't see the benefit. We'd need to account for every possible extra argument in our base listener, which is very messy. It would include
- position
- range
- trigger character
- uri

And possibly more in the future. It would also be confusing for extension authors. If they wanted to override `initialize` to carry some state, they would need to always include these arguments even if it's not related to the request they are extending. E.g.: why would you need to accept trigger character for semantic highlighting?

Also each request has a different response, which is why we always override `initialize`. For example, `hover` returns a single object as the response, but `document symbol` returns an array. If we initialized the response in the parent class as an array for all requests, it would be inconvenient for requests like `hover`. If we initialized it as `nil`, it's inconvenient for requests that return arrays.

I believe keeping consistency only for same request listeners will make the code more organized. Of course, we can always revisit if we conclude otherwise.